### PR TITLE
consistency fixes

### PR DESCRIPTION
**Pull request by monoclex**
elaborate on "The type."
add some generic overloads for things without generic overloads
.gitignore the .vs folder
consistently put => on a newline

### CODE REVIEW

File: StrictEmit/ILGeneratorExtensions.Calls.cs [hunk 1]
```diff
@@ -268,17 +268,35 @@ public static void EmitCallVirtual([NotNull] this ILGenerator il, [NotNull] Cons
         /// <summary>
         /// Attempts to cast an object passed by reference to the specified class.
         /// </summary>
+        /// <typeparam name="T">The type to cast the highest item on the stack to.</typeparam>
```
Comment (Nihlus) [inline StrictEmit/ILGeneratorExtensions.Calls.cs]: This should be "on top of the stack" to use proper stack lingo. Same for the others.

File: StrictEmit/ILGeneratorExtensions.Objects.cs [hunk 2]
```diff
@@ -30,7 +30,7 @@ public static partial class ILGeneratorExtensions
         /// Creates a new object or a new instance of a value type, pushing an object reference
         /// (type <strong>O</strong>) onto the evaluation stack.
         /// </summary>
-        /// <typeparam name="T">The type.</typeparam>
+        /// <typeparam name="T">The type of object to emit.</typeparam>
```
Comment (Nihlus) [inline StrictEmit/ILGeneratorExtensions.Objects.cs]: We're not emitting an object, we're emitting an opcode. Perhaps "to create an instance of"? Same for the others.

File: StrictEmit/ILGeneratorExtensions.Objects.cs [hunk 3]
```diff
@@ -96,16 +97,17 @@ public static void EmitSetObject([NotNull] this ILGenerator il, [NotNull] Type o
...
-        public static void EmitIsInstance<T>([NotNull] this ILGenerator il) => il.EmitIsInstance(typeof(T));
+        public static void EmitIsInstance<T>([NotNull] this ILGenerator il)
+            => il.EmitIsInstance(typeof(T));
 
         /// <summary>
         /// Tests whether an object reference (type <strong>O</strong>) is an instance of a particular class.
         /// </summary>
         /// <param name="il">The generator where the IL is to be emitted.</param>
-        /// <param name="typeToCheck">The type.</param>
+        /// <param name="typeToCheck">The type of object to emit if it is an instance of.</param>
```
Comment (Nihlus) [inline StrictEmit/ILGeneratorExtensions.Objects.cs]: Rather clunky. Perhaps "The object type to test against"?

File: StrictEmit/ILGeneratorExtensions.Utility.cs [hunk 4]
```diff
@@ -30,46 +30,61 @@ public static partial class ILGeneratorExtensions
...
         [PublicAPI]
-        public static void EmitPop([NotNull] this ILGenerator il) => il.Emit(OpCodes.Pop);
+        public static void EmitPop([NotNull] this ILGenerator il)
+            => il.Emit(OpCodes.Pop);
 
         /// <summary>
         /// Pushes the size, in bytes, of a supplied value type onto the evaluation stack.
         /// </summary>
-        /// <typeparam name="T">The type.</typeparam>
+        /// <typeparam name="T">The type to emit the size of in bytes.</typeparam>
```
Comment (Nihlus) [inline StrictEmit/ILGeneratorExtensions.Utility.cs]: Also a little clunky. Perhaps "to calculate the size of"? "in bytes" is not required, we specify it above. Same for the others.

File: StrictEmit/ILGeneratorExtensions.Values.cs [hunk 5]
```diff
@@ -256,18 +274,19 @@ public static void EmitLoadToken([NotNull] this ILGenerator il, [NotNull] FieldI
         /// Converts a <see cref="Type"/> token to its runtime representation, pushing it onto the evaluation stack.
         /// </summary>
         /// <param name="il">The generator where the IL is to be emitted.</param>
-        /// <param name="type">The type.</param>
+        /// <param name="type">The type to convert to it's runtime representation.</param>
```
Comment (Nihlus) [inline StrictEmit/ILGeneratorExtensions.Values.cs]: its, not it's.

**Review by Nihlus — CHANGES_REQUESTED** [review]

File: StrictEmit/ILGeneratorExtensions.Objects.cs (hunk 3 above)
Comment (monoclex) [reply StrictEmit/ILGeneratorExtensions.Objects.cs]: i just noticed that right above you have "The type to emit the instance of instruction", should i use that instead?

Comment (Nihlus) [reply StrictEmit/ILGeneratorExtensions.Objects.cs]: No, I don't particularly like that either. Let's change both.

File: StrictEmit/ILGeneratorExtensions.Values.cs [hunk 6]
```diff
@@ -281,10 +300,11 @@ public static void EmitUnbox([NotNull] this ILGenerator il, [NotNull] Type type)
         /// <summary>
         /// Converts the boxed representation of a type specified in the instruction to its unboxed form.
         /// </summary>
-        /// <typeparam name="T">The type.</typeparam>
+        /// <typeparam name="T">The type to unbpx.</typeparam>
```
Comment (Nihlus) [inline StrictEmit/ILGeneratorExtensions.Values.cs]: Typo: unbpx

**Review by Nihlus — CHANGES_REQUESTED** [review]

**Comment by Nihlus** [comment]
Superb work :+1: Just one final thing and we're good to merge.

File: StrictEmit/ILGeneratorExtensions.Values.cs (hunk 6 above)
Comment (monoclex) [reply StrictEmit/ILGeneratorExtensions.Values.cs]: lol, fixed

**Review by Nihlus — APPROVED** [review]

**Pull request merged**